pkg/management/postgres/metrics/collector.go

```diff
@@ -50,33 +50,48 @@ type QueriesCollector struct {
 
 	errorUserQueries      *prometheus.CounterVec
 	errorUserQueriesGauge prometheus.Gauge
+
+	computedMetrics []prometheus.Metric
 }
 
 // Name returns the name of this collector, as supplied by the user in the configMap
 func (q QueriesCollector) Name() string {
 	return q.collectorName
 }
 
+// GetComputedMetricsCount returns the number of constant Prometheus metrics that
+// are updated and ready for collection
+func (q QueriesCollector) GetComputedMetricsCount() int {
+	return len(q.computedMetrics)
+}
+
 var isPathPattern = regexp.MustCompile(`[][*?]`)
 
-// Collect loads data from the actual PostgreSQL instance
-func (q QueriesCollector) Collect(ch chan<- prometheus.Metric) error {
+// Update recomputes the metrics from the user queries
+func (q *QueriesCollector) Update() error {
 	// Reset before collecting
 	q.errorUserQueries.Reset()
 
-	err := q.collectUserQueries(ch)
+	err := q.createMetricsFromUserQueries()
 	if err != nil {
 		return err
 	}
+	return nil
+}
+
+// Collect sends the updated values to the output channel
+func (q QueriesCollector) Collect(ch chan<- prometheus.Metric) {
+	q.collectComputedMetrics(ch)
 
 	// Add errors into errorUserQueriesVec and errorUserQueriesGauge metrics
 	q.errorUserQueriesGauge.Collect(ch)
 	q.errorUserQueries.Collect(ch)
-
-	return nil
 }
 
-func (q *QueriesCollector) collectUserQueries(ch chan<- prometheus.Metric) error {
+func (q *QueriesCollector) createMetricsFromUserQueries() error {
+	// refresh
+	q.computedMetrics = []prometheus.Metric{}
```

Comment:
Do we need to lock before we write? 🤔

Reply:
well, only this function updates that, not sure if it's necessary, but might be a good thing